src/sentry/grouping/enhancer/__init__.py

```diff
@@ -287,7 +299,8 @@ def as_dict(self, with_rules=False):
             rv["rules"] = [x.as_dict() for x in self.rules]
         return rv
 
-    def _to_config_structure(self):
+    def _to_config_structure(self) -> list[Any]:
+        # TODO: Can we switch this to a tuple so we can type it more exactly?
```

Comment:
Maybe a named tuple?

```python
        from collections import namedtuple
        
        ConfigStructure = namedtuple('ConfigStructure', ['version', 'bases', 'rules'])

        return ConfigStructure(
            version=self.version,
            bases=self.bases,
            rules=[x._to_config_structure(self.version) for x in self.rules],
        )
```

Reply:
Yup, that'd work. I'd rather leave it for a future PR, though, as my goal with this one is purely to add types, not change behavior.

Reply:
Works for me.